src/Microsoft.OData.Core/ODataWriterCore.cs

```diff
@@ -4133,7 +4133,7 @@ internal ResourceBaseScope(WriterState state, ODataResourceBase resource, ODataR
 
                 if (resource != null)
                 {
-                    duplicatePropertyNameChecker = writerSettings.Validator.CreateDuplicatePropertyNameChecker();
+                    duplicatePropertyNameChecker = writerSettings.Validator.GetDuplicatePropertyNameChecker();

```

Comment:
Is this what gets returned in the `ODataJsonLightWriter`?

Reply:
Yes, in the `EndResource(ODataResource resource)` method.